sklearn/utils/validation.py

```diff
@@ -866,21 +865,18 @@ def check_symmetric(array, tol=1E-10, raise_warning=True,
     return array
 
 
-def check_is_fitted(estimator, attributes, msg=None, all_or_any=all):
+def check_is_fitted(estimator, attributes='deprecated', msg=None):
```

Comment:
The behavior is already not back-compatible. Since we mention in the documentation that these utils can change from a version to another, I would not bother with a deprecation warning for the `attributes` parameters knowing that one can have some side-effect with `all_or_any`.

Reply:
How is it not backward compatible? Oh I could deprecate ``all_or_any`` as well?

Reply:
If somebody is using `all_or_any` now, nothing would happen and this is not an attribute of the function as well. But as I mentioned, we clearly state in the documentation that `utils` are not following the deprecation cycle and can change: https://scikit-learn.org/stable/developers/utilities.html

Reply:
True, only deprecating one doesn't make sense. But also see the discussion at #6616. Basically, the docs say that but people ignore it and it might not be good if we enforce it and should make things private instead.

Reply:
OK I see. I am sure I was one of these people that complain at least once (then @lesteve show me the red box :))
I really feel that having the utils private could help to move quickly sometimes and help third-party project (at the cost of potential breaking if they use them). So deprecation it is :)

Reply:
so should I add one for ``all_or_any`` then?